packages/mathtype-froala3/wiris.src.js

```diff
@@ -294,19 +292,19 @@ export class FroalaIntegration extends IntegrationModel {
     // Value can be undefined.
     if (selectedImage) {
       if (($btn.parent()[0].hasAttribute('class') && $btn.parent()[0].getAttribute('class').indexOf('fr-buttons') === -1) || (selectedImage[0]
-                && ($(selectedImage[0]).hasClass(Configuration.get('imageClassName')) || $(selectedImage[0]).contents().hasClass(Configuration.get('imageClassName'))))) { // Is a MathType image.
+                && ((selectedImage[0]).classList.contains(Configuration.get('imageClassName')) || (selectedImage[0]).contents().classList.contains(Configuration.get('imageClassName'))))) { // Is a MathType image.
         // Show MathType icons if previously were hiden.
         $btn.removeClass('fr-hidden');
         // Disable resize box.
-        if (!$('#wrs_style').get(0)) { // eslint-disable-line no-undef
-          $('head').append('<style id="wrs_style">.fr-image-resizer {pointer-events: none;}</style>');
+        if (!document.getElementById('wrs_style')) { // eslint-disable-line no-undef
+          (document.getElementsByTagName('head')[0]).append('<style id="wrs_style">.fr-image-resizer {pointer-events: none;}</style>');
```

Comment:
Same thing about the parentheses, check if they can be dropped for better readability.

Reply:
> Good job, Carla! Removing the jQuery dependency from the Froala package is such a good move. I am happy that you decide to get rid of this dependency. partying_face
> 
> The reason I am asking for review is that I believe that there are other uses of the jQuery's `hasClass` function on other parts of the `mathtype-froala3` plugin.
> 
> Like on these next lines:
> 
> * packages/mathtype-froala3/wiris.src.js:280
> * packages/mathtype-froala3/wiris.src.js:327
> 
> I believe you should remove this function calls to in order to avoid errors.

You're right, i've changed them and tested they work properly

> At the same time, I've found that the `mathtype-froala` plugin also holds references to this same jQuery function.
> 
> * packages/mathtype-froala/wiris.src.js:279
> * packages/mathtype-froala/wiris.src.js:295
> * packages/mathtype-froala/wiris.src.js:327
> 
> Since the current installation instructions don't mention that the jQuery dependency needs to be added manually and it's not included on the package dependency, I guess this could be causing unexpected behaviors in some use cases.
> 
> Would you mind to have a look to it, on this package, too?

Regarding the Froala 2 package, froala 2 has jQuery as a dependency, the froala team removed this when updating to V3 and above, so jQuery dependencies and functions on this package are necessary

@dcanet-at-wiris

Reply:
Right! I'm on it